### PR TITLE
[go-gen] Generate handler.go for Auth service aside from Init function

### PR DESCRIPTION
**Pull request by willjvsmith**
* Moves `generateInit` for the `comm/handler.go`s to `GoCommonGenerator`
  * At the moment `GoCommonGenerator` is small, but if it gets bigger I'll split it up into `Main`, `Comm`, `DAO`, just as I have done for the other generation

The `init` function is not generated in this PR, as I want to use the common function, but `init` function has changed since the last release. Therefore changing it will break the old generation and their accompanying integration tests. Best do it in a separate PR.

### CODE REVIEW

File: src/main/resources/go/genFiles/auth/comm/create_jwt_credential.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	}
+
+	return &jwt, nil
+}
+
+// CreateJWTCredential provisions a new HS256 JWT credential
+func (comm *Handler) CreateJWTCredential() (*JWTCredential, error) {
+	hostname, ok := comm.Services["kong-admin"]
+	if !ok {
+		return nil, fmt.Errorf("service %s's hostname not in config file", "kong-admin")
```
Comment (jaylees14) [inline src/main/resources/go/genFiles/auth/comm/create_jwt_credential.go]: Can we remove this unnecessary `%s`

File: src/main/scala/temple/generate/service/go/GoServiceGenerator.scala [hunk 2]
```diff
@@ -55,7 +55,7 @@ object GoServiceGenerator extends ServiceGenerator {
         GoCommonGenerator.generatePackage("comm"),
         GoServiceCommGenerator.generateImports(serviceRoot.module),
         GoServiceCommGenerator.generateStructs(),
-        GoServiceCommGenerator.generateInit(),
+        GoCommonGenerator.generateCommInit(),
```
Comment (jaylees14) [inline src/main/scala/temple/generate/service/go/GoServiceGenerator.scala:58]: I'm struggling to see how a `CommInit` can be `Common`?

Comment (willjvsmith) [reply src/main/scala/temple/generate/service/go/GoServiceGenerator.scala:58]:
```
// Init sets up the Handler object with a list of services from the config
func Init(config *util.Config) *Handler {
	return &Handler{config.Services}
}
```

This is the same regardless of the service, including Auth, so can use the same function to read the same file.

Comment (jaylees14) [reply src/main/scala/temple/generate/service/go/GoServiceGenerator.scala:58]: I see, that was quite hard to follow from the naming

File: src/main/resources/go/genFiles/auth/comm/create_jwt_credential.go (hunk 1 above)
Comment (willjvsmith) [reply src/main/resources/go/genFiles/auth/comm/create_jwt_credential.go]: Needs to be removed spec-golang too, will make a card with an update for it

File: src/main/scala/temple/generate/service/go/GoServiceGenerator.scala (hunk 2 above)
Comment (jaylees14) [reply src/main/scala/temple/generate/service/go/GoServiceGenerator.scala:58]: `GoCommon` sounds like it can be used anywhere, whereas I think what you want is `GoCommGenerator`, which then gets extended by `GoServiceCommGenerator` and `GoAuthCommGenerator`

**Review by jaylees14 — APPROVED** [review]

File: src/main/scala/temple/generate/service/go/GoServiceGenerator.scala (hunk 2 above)
Comment (willjvsmith) [reply src/main/scala/temple/generate/service/go/GoServiceGenerator.scala:58]: See PR description, will address in next PR

**Review by Lewiky — APPROVED** [review]
